src/backend/metal/src/device.rs

```diff
@@ -290,13 +287,21 @@ impl hal::PhysicalDevice<Backend> for PhysicalDevice {
                 return None
             }
         }
-        let height = if dimensions >= 2 { 4096 } else { 1 };
-        let depth = if dimensions >= 3 { 4096 } else { 1 };
-        let max_dimension = 4096f32; // Max of {width, height, depth}
+        if dimensions == 1 && usage.intersects(image::Usage::COLOR_ATTACHMENT | image::Usage::DEPTH_STENCIL_ATTACHMENT) {
+            // MTLRenderPassDescriptor texture must not be MTLTextureType1D
+            return None;
+        }
+        //TODO: actually query this data
+        let max_dimension = 4096u32;
+        let max_extent = image::Extent {
+            width: max_dimension,
+            height: if dimensions >= 2 { max_dimension } else { 1 },
+            depth: if dimensions >= 3 { max_dimension } else { 1 },
+        };
 
         self.private_caps.map_format(format).map(|_| image::FormatProperties {
-            max_extent: image::Extent { width, height, depth },
-            max_levels: max_dimension.log2().ceil() as u8 + 1,
+            max_extent,
+            max_levels: if dimensions == 1 { 1 } else { 12 },
```

Comment:
Why `12`?

Reply:
based on `max_dimension`. I should make it cleaner...

Reply:
Oh ok, fair enough 👍 If we query this somehow we'll need to move back to the calculation though, or precompute all of these.